mozetl/bhr_collection/bhr_collection.py

```diff
@@ -0,0 +1,1359 @@
+                "isSplit": True,
+            },
+            "file_data": [
+                [
+                    (t["name"] + "_" + k, v)
+                    for k, v in self.process_thread(t).iteritems()
+                    if k != "name"
+                ]
+                for t in self.thread_table.get_items()
+            ],
+        }
+
+    def process_into_profile(self):
+        print("Processing into final format...")
+        if self.config["split_threads_in_out_file"]:
+            return [
+                {
+                    "name": t["name"],
+                    "threads": [self.process_thread(t)],
+                    "usageHoursByDate": self.usage_hours_by_date,
+                    "uuid": self.config["uuid"],
+                }
+                for t in self.thread_table.get_items()
+            ]
+
+        return {
+            "threads": [self.process_thread(t) for t in self.thread_table.get_items()],
+            "usageHoursByDate": self.usage_hours_by_date,
+            "uuid": self.config["uuid"],
+        }
+
+
+def deep_merge(original, overrides):
+    original_copy = original.copy()
+    for k, v in overrides.iteritems():
+        if (
+            isinstance(v, dict)
+            and k in original_copy
+            and isinstance(original_copy[k], dict)
+        ):
+            original_copy[k] = deep_merge(original_copy[k], v)
+        else:
+            original_copy[k] = v
+    return original_copy
+
+
+def shallow_merge(original, overrides):
+    original_copy = original.copy()
+    for k, v in overrides.iteritems():
+        original_copy[k] = v
+    return original_copy
+
+
+def time_code(name, callback):
+    print("{}...".format(name))
+    start = time.time()
+    result = callback()
+    end = time.time()
+    delta = end - start
+    print("{} took {}ms to complete".format(name, int(round(delta * 1000))))
+    debug_print_rdd_count(result)
+    return result
+
+
+def get_prop(val, prop):
+    if val is None:
+        return None
+
+    for key in prop.split("/"):
+        val = val[key]
+        if val is None:
+            return None
+    return val
+
+
+def get_ping_properties(ping, properties):
+    result = {}
+    for prop in properties:
+        result[prop] = get_prop(ping, prop)
+    return result
+
+
+def properties_are_not_none(ping, properties):
+    for prop in properties:
+        if ping[prop] is None:
+            return False
+
+    return True
+
+
+def get_data(sc, sql_context, config, date, end_date=None):
+    sql_context.sql("set spark.sql.shuffle.partitions={}".format(sc.defaultParallelism))
+
+    if end_date is None:
+        end_date = date
+
+    submission_start_str = date - timedelta(days=5)
+    submission_end_str = end_date + timedelta(days=5)
+
+    date_str = date.strftime("%Y%m%d")
+    end_date_str = end_date.strftime("%Y%m%d")
+
+    pings_df = (
+        spark.read.format("bigquery")
+        .option("table", "moz-fx-data-shared-prod.telemetry_stable.bhr_v4")
```

Comment:
Is this using the new structured format as introduced in https://github.com/mozilla-services/mozilla-pipeline-schemas/pull/636 on 2020-12-07? From what I'm reading, it looks like it does due to the way that the stacks are being parsed, but I just want to make sure.

Reply:
Yes this is what broke the job last week and needed to get fixed before moving it over.